apps/admin/backend/src/store.ts

```diff
@@ -966,14 +973,98 @@ export class Store {
         batchId: row.batchId,
         scannerId: row.scannerId,
         precinctId: row.precinctId,
-        card: row.sheetNumber
-          ? { type: 'hmpb', sheetNumber: row.sheetNumber }
-          : { type: 'bmd' },
+        card: this.convertSheetNumberToCard(row.sheetNumber),
         votes: JSON.parse(row.votes),
       };
     }
   }
 
+  /**
+   * Gets card tallies grouped by cast vote record attributes.
+   */
+  *getCardTallies({
+    electionId,
+    election,
+    groupBy = {},
+  }: {
+    electionId: Id;
+    election: Election;
+    groupBy?: Tabulation.GroupBy;
+  }): Generator<Tabulation.GroupOf<CardTally>> {
+    const cvrSelectParts: string[] = [];
+    const groupByParts: string[] = [];
+
+    if (groupBy.groupByBallotStyle || groupBy.groupByParty) {
+      cvrSelectParts.push('cvrs.ballot_style_id as ballotStyleId');
+      groupByParts.push('cvrs.ballot_style_id');
+    }
+
+    if (groupBy.groupByBatch) {
+      cvrSelectParts.push('cvrs.batch_id as batchId');
+      groupByParts.push('cvrs.batch_id');
+    }
+
+    if (groupBy.groupByPrecinct) {
+      cvrSelectParts.push('cvrs.precinct_id as precinctId');
+      groupByParts.push('cvrs.precinct_id');
+    }
+
+    if (groupBy.groupByScanner) {
+      cvrSelectParts.push('scanner_batches.scanner_id as scannerId');
```

Comment:
nit: should this just be `selectParts` if it includes fields from both `cvrs` and `scanner_batches`?

Reply:
I don't have an meaningful opinion one way or another (in other places, I'm treating `scannerId` as a "CVR attribute" but in the SQL context I think it's a fair clarification), so switched it - cb4f2153a.